File: LeetCode/Array/Python/Leetcode1417-reformat-the-string.py
class Solution:
    def reformat(self, s: str) -> str:
        cnt,n = 0,len(s)
        for i in range(n):
            if s[i].isdigit(): cnt += 1
        if abs(cnt - (n - cnt)) > 1: return ''
        flag = cnt > (n - cnt)
        s = list(s)
        j = 1
        for i in range(0,n,2):
            if s[i].isdigit() != flag:
                while s[j].isdigit() != flag: j += 2
                s[i],s[j] = s[j],s[i]
        return ''.join(s)

File: LeetCode/Array/Golang/Leetcode1417-reformat-the-string.go
package sbgo

import "unicode"

func reformat(s string) string {
	cnt, n := 0, len(s)
	for _, ch := range s {
		if unicode.IsDigit(ch) {
			cnt++
		}
	}
	if abs(cnt+cnt-n) > 1 {
		return ""
	}
	flag := cnt > (n - cnt)
	arr := []byte(s)
	for i, j := 0, 1; i < n; i += 2 {
		if unicode.IsDigit(rune(arr[i])) != flag {
			for unicode.IsDigit(rune(arr[j])) != flag {
				j += 2
			}
			arr[i], arr[j] = arr[j], arr[i]
		}
	}
	return string(arr)
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
